### PR TITLE
python,python3: Fix CVE-2019-9948 - local_file:// allowed in urllib

### PR DESCRIPTION
**Pull request by jefferyto**
Maintainer: me, @commodo 
Compile tested: armvirt-64, 2019-06-03 snapshot sdk
Run tested: none

Description:
These patches address issue:
CVE-2019-9948: Unnecessary URL scheme exists to allow local_file:// reading file in urllib

Link to Python issue:
https://bugs.python.org/issue35907

Issue 35907 is still currently open, waiting for a decision for Python 3.5; these patches for Python 2.7 and 3.7 have been merged.

Signed-off-by: Jeffery To <jeffery.to@gmail.com>

### CODE REVIEW

File: lang/python/python/patches/024-bpo-35907-Complete-test_urllib.test_local_file_open-GH-13506.patch [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ 
+ # Just commented them out.
+ # Can't really tell why keep failing in windows and sparc.
+diff --git a/Misc/NEWS.d/next/Library/2019-02-13-17-21-10.bpo-35907.ckk2zg.rst b/Misc/NEWS.d/next/Library/2019-02-13-17-21-10.bpo-35907.ckk2zg.rst
+index bb187d8d65a5..6a448ce6261c 100644
+--- a/Misc/NEWS.d/next/Library/2019-02-13-17-21-10.bpo-35907.ckk2zg.rst
++++ b/Misc/NEWS.d/next/Library/2019-02-13-17-21-10.bpo-35907.ckk2zg.rst
+@@ -1 +1,3 @@
+-CVE-2019-9948: Avoid file reading as disallowing the unnecessary URL scheme in urllib.urlopen
++CVE-2019-9948: Avoid file reading as disallowing the unnecessary URL scheme in
```
Comment (commodo) [inline lang/python/python/patches/024-bpo-35907-Complete-test_urllib.test_local_file_open-GH-13506.patch]: i would not add the patches don't change lib functionality;
they are just noise
a lot [or all] of the tests are not even packaged;

Comment (commodo) [reply lang/python/python/patches/024-bpo-35907-Complete-test_urllib.test_local_file_open-GH-13506.patch]: i am mostly referring to patch `[PATCH] bpo-35907: Complete test_urllib.test_local_file_open()`

Comment (jefferyto) [reply lang/python/python/patches/024-bpo-35907-Complete-test_urllib.test_local_file_open-GH-13506.patch]: Actually I'd like to see the tests packaged at some point (for both the Python interpreter and for libraries), but this is just a far away goal. I'm ok with removing the tests patch.

You are ok with the patches that only affect NEWS? :joy:

Comment (commodo) [reply lang/python/python/patches/024-bpo-35907-Complete-test_urllib.test_local_file_open-GH-13506.patch]: > You are ok with the patches that only affect NEWS? joy

I'd remove them if they only affect NEWS.
We will get them eventually when we upgrade.
Thing is, I would not add stuff (as patches) if we get them as upgrade.

Comment (jefferyto) [reply lang/python/python/patches/024-bpo-35907-Complete-test_urllib.test_local_file_open-GH-13506.patch]: Ok :smile:

**Comment by jefferyto** [comment]
I've removed some of the patches that only change tests or NEWS.

**Comment by commodo** [comment]
Thanks :)
I will admit it's a bit OCD-related to patch things at a minimum.
But I also found that it keeps things easier to see.

**Comment by commodo** [comment]
LGTM

**Pull request merged**